tests/lib/thrift/ThriftTest/src/client.cpp

```diff
@@ -0,0 +1,509 @@
+/*
+ * Copyright 2022 Young Mei
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#include <array>
+#include <cmath>
+#include <limits>
+#include <locale>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+
+#include <pthread.h>
+#include <signal.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <thrift/protocol/TBinaryProtocol.h>
+#include <thrift/server/TSimpleServer.h>
+#include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TSocket.h>
+
+#include "ThriftTest.h"
+
+#include "TestUtils.hpp"
+
+using namespace std;
+
+using namespace apache::thrift;
+using namespace apache::thrift::async;
+using namespace apache::thrift::concurrency;
+using namespace apache::thrift::protocol;
+using namespace apache::thrift::transport;
+using namespace apache::thrift::server;
+using namespace thrift::test;
+
+#define HELLO_SERVER_STACK_SIZE 4096
+static K_THREAD_STACK_DEFINE(ThriftTest_server_stack, HELLO_SERVER_STACK_SIZE);
+
+struct ctx {
+  shared_ptr<ThriftTestClient> client;
+  shared_ptr<TServer> server;
+  pthread_t server_thread;
+};
+static ctx context;
+
+shared_ptr<TServer> setup_server(const std::string& my_addr, const int port);
+
+static void init_Xtruct(Xtruct& s);
+
+static void binary_fill(std::string& str, string::size_type siz);
+
+static uint64_t now();
+
+static void void_test(void) {
+  ThriftTestClient& testClient = *(context.client);
```

Comment:
Why not just use `context.client->testVoid()` in this function? It's a single line of code and achieves the same results.